UI/Layers/LayerConfiguration/LayerCompositionWidget.gd

```diff
@@ -25,34 +25,53 @@ func _ready():
 	reload_button.connect("pressed",Callable(self,"_on_layer_reload_pressed"))
 	edit_window.connect("change_color_tag",Callable(self,"_change_color_tag"))
 	edit_window.connect("translate_to_layer",Callable(self,"_emit_translate_to_layer"))
+	$VBox/EditMenu.connect("ui_geolayers_set_visible",Callable(self,"_ui_geolayers_set_visible"))
 	visibility_button.connect("toggled",Callable(self,"_layer_change_visibility"))
 	layer_composition.connect("layer_changed",Callable(self,"_reload"))
-
+	$GeoLayers.connect(
+		"item_collapsed", 
+		func(treeitem: TreeItem): 
+			if treeitem.is_any_collapsed(): $GeoLayers.set_custom_minimum_size(Vector2i(0, 8))
+			else: $GeoLayers.set_custom_minimum_size(Vector2i(0, 80)))
 
 func _reload():
-	# FIXME: check wheter raster of rfeature composition 
-	#if layer_composition is LayerComposition:
-	#	icon.texture = load("%s/raster.svg" % icon_prefix)
-	#elif layer_composition is FeatureLayerComposition:
-	#	icon.texture = load("%s/vector.svg" % icon_prefix)
-	#else:
-	match layer_composition.render_type:
-		LayerComposition.RenderType.REALISTIC_TERRAIN:
-			icon.texture = load("%s/world.svg" % icon_prefix)
-		LayerComposition.RenderType.BASIC_TERRAIN:
-			icon.texture = load("%s/world.svg" % icon_prefix)
-		LayerComposition.RenderType.TWODIMENSIONAL:
-			icon.texture = load("%s/map.svg" % icon_prefix)
-		LayerComposition.RenderType.VEGETATION:
-			icon.texture = load("%s/grass.svg" % icon_prefix)
-		_:
-			icon.texture = load("%s/layer.svg" % icon_prefix)
+	if layer_composition.render_info is LayerComposition.ObjectRenderInfo:
+		icon.texture = load("%s/vector.svg" % icon_prefix)
+	else:
+		icon.texture = load("%s/raster.svg" % icon_prefix)
+	# FIXME: Reimplement with render_info
```

Comment:
What to do with that? Let's not leave those things in a FIXME state for 2 years :)